pkg/core/combat/enemy.go

```diff
@@ -72,8 +72,8 @@ func (c *Handler) EnemyByDistance(x, y float64, excl TargetKey) []int {
 		if v.Key() == excl {
 			continue
 		}
-		vx, vy := v.Shape().Pos()
-		dist := math.Pow(x-vx, 2) + math.Pow(y-vy, 2)
+		vPos := v.Shape().Pos()
+		dist := math.Pow(p.X-vPos.X, 2) + math.Pow(p.Y-vPos.Y, 2)
```

Comment:
nit: operations like this we might want to generalize in point IE add something like:
```go
func (p Point) MagnitudeSquared() float64 {
  return p.X*p.X + p.Y*p.Y
}

func (p Point) Magnitude() float64 {
  return math.Sqrt(p.MagnitudeSquared())
}
```

Then the logic here can be simplified to:
```go
dist := p.Sub(vPos).MagnitudeSquared()
```

Might also help to prevent bugs where people use the squared magnitude but do not compare to a squared radius